cranelift-codegen/src/simple_gvn.rs

```diff
@@ -124,7 +124,8 @@ pub fn do_simple_gvn(func: &mut Function, domtree: &mut DominatorTree) {
             use crate::scoped_hash_map::Entry::*;
             match visible_values.entry(key) {
                 Occupied(entry) => {
-                    debug_assert!(domtree.dominates(*entry.get(), inst, &func.layout));
+                    let dominates = domtree.dominates(*entry.get(), inst, &func.layout);
+                    debug_assert!(dominates);
```

Comment:
Out of curiosity, why was it split into two lines? It seems `dominates` would be dead in a non-debug build.

Reply:
clippy was complaining about using something mutable inside the `debug_assert!` (IIRC it was `func.layout`); moving the call out of the macro fixed the warning but honestly I didn't give it too much thought. Is there a better way?

Reply:
Seems like a false positive to me, see [this reduced case](https://github.com/rust-lang/rust-clippy/issues/4737#issuecomment-573296739), and it's probably better to ignore clippy here if it's wrong.